corehq/apps/app_manager/util.py

```diff
@@ -119,37 +119,33 @@ def split_path(path):
 
 
 def save_xform(app, form, xml):
-    def change_xmlns(xform, replacing):
+    from corehq.apps.app_manager.models import ShadowForm
+    def change_xmlns(xform, old_xmlns, new_xmlns):
         data = xform.data_node.render()
-        xmlns = "http://openrosa.org/formdesigner/%s" % form.get_unique_id()
-        data = data.replace(replacing, xmlns, 1)
+        data = data.replace(old_xmlns, new_xmlns, 1)
         xform.instance_node.remove(xform.data_node.xml)
         xform.instance_node.append(parse_xml(data))
-        xml = xform.render()
-        return xform, xml
+        return xform.render()
 
     try:
         xform = XForm(xml)
     except XFormException:
         pass
     else:
-        duplicates = app.get_xmlns_map()[xform.data_node.tag_xmlns]
-        for duplicate in duplicates:
-            if form == duplicate:
-                continue
-            else:
-                xform, xml = change_xmlns(xform, xform.data_node.tag_xmlns)
-                break
```

Comment:
Good point. I just removed the thing that changes the XMLNS if it does not match. I'm not sure if that's ever necessary anyway.